Latest changes ER-97, ER-98 and ER-99

jdiazarg opened the pull request:
- changes to complete ER-97, ER-98 and ER-99
- problems with frisby, needs to solve the mac problem. To test admin/UserAjax we dont have a kind of setUp() before the each test to login in session first. Its needed in admin/UserAjax mock checkAuth().

- in the future if we want manage many auth levels for one user, we need upgrade Roles as a relationship between entity User and Role.

Commit:
fixed User model getRoles to use with authenticator. Updated UserAjax and admin\UserAjax controllers.

Commit:
Added changes to action list(). removed getAdmins() in user model and tests.

Commit:
new Mailgun model, modified forgotPassword() action.

Commit:
changed to email view text.

Commit:
moved Mailgun to Users project. Added changes to email view and Mailgun model.

Commit:
added mailgun config file.

saarmstrong commented on src/controllers/UserAuthenticationAjax.php:
```diff
@@ -0,0 +1,256 @@
+        );
+
+        try {
+            $data = json_decode(file_get_contents("php://input"));
```
this does not always pick up POST/GET params. to see an example of this, use POSTman or just make a get request manually in the browser to see the lack of query string values.

please check all instances where we get params this way and make sure we also check the _REQUEST array (and strip tags from) to make sure we get data.

saarmstrong commented on src/controllers/UserAjax.php:
```diff
@@ -297,10 +297,10 @@ public function getGetusers()
 
     }
 
-    public function getGetUser()
```
make sure you merge in the latest updates from the current `develop` branch. I think a change I made last week may get clobbered by this PR

saarmstrong commented on src/controllers/admin/UserAjax.php:
```diff
@@ -329,12 +335,15 @@ public function getRetrieve()
-                              'last_login' => $user->getLastLogin(),
-                              'gateway_customer_id'=> $user->getGatewayCustomerId()
+                            'email'    => $user->getEmail(),
+                            'role'     => $user->getRole(),
```
we should have this return an object or array including the Role ID and the user friendly Role Name string.

saarmstrong reviewed (changes requested):
Couple of things: 

- We need a unit test for the mailgun model
- Please see my notes about the query string params getting dropped with the `file_get_contents` method of param retrieval. This works with some methods, but not all, and def looses anything passed by query string. If you need help reproducing this, attempt to make a manual request in the browser and note that anything passed into the query string is absent from the params list
- Please delete the `getRead` method. This is not even used?
- Please add better doc blocks to your unit tests and explain what you are testing
- Please complete the doc blocks in ALL the model methods and explain what the methods are doing.
- Make sure you abide by our company's syntax and styling guidelines. I def see some improvement, but I am not seeing a carriage return after some method definitions and I def want to make sure these classes are PRISTINE examples of good styling and documentation. If you need help setting up phplint, please hit me up on slack.

Commit:
added new test suit. Fixed forgotPassword() pass newpass is not useful.

Commit:
- changes in code using styling guides. (almost human posible using phplint)
- added doc block and comments in test cases and models.
- removed getRead() from controllers.
- in post controller actions, check the $_POST array in case that data is empty.

Commit:
more styling guides.
